spec/lib/appsignal/probes/gvl_spec.rb

```diff
@@ -84,4 +104,64 @@ def gauges_for(metric)
       expect(gauges_for("gvl_waiting_threads")).to be_empty
     end
   end
+
+  context "when the process name is a custom value" do
+    before(:each) do
+      FakeGVLTools::WaitingThreads.enabled = true
+    end
+
+    it "uses only the first word as the process name" do
+      $PROGRAM_NAME = "sidekiq 7.1.6 app [0 of 5 busy]"
+      probe.call
+
+      expect(gauges_for("gvl_waiting_threads")).to eq [
+        [0, {
+          :hostname => hostname,
+          :process_name => "sidekiq",
+          :process_id => Process.pid
+        }],
+        [0, { :hostname => hostname }]
+      ]
+    end
+  end
+
+  context "when the process name is a path" do
+    before(:each) do
+      FakeGVLTools::WaitingThreads.enabled = true
+    end
+
+    it "uses only the binary name as the process name" do
+      $PROGRAM_NAME = "/foo/folder with spaces/bin/rails"
+      probe.call
+
+      expect(gauges_for("gvl_waiting_threads")).to eq [
+        [0, {
+          :hostname => hostname,
+          :process_name => "rails",
+          :process_id => Process.pid
+        }],
+        [0, { :hostname => hostname }]
+      ]
+    end
+  end
+
+  context "when the process name is an empty string" do
+    before(:each) do
```

Comment:
```suggestion
    before do
```
Each is the default and doesn't need to specified. (Also in the other tests.)